packages/photon/src/index.js

```diff
@@ -98,10 +101,16 @@ function isAlreadyPhotoned( host ) {
  * Determine which Photon server to connect to: `i0`, `i1`, or `i2`.
  *
  * Statically hash the subdomain based on the URL, to optimize browser caches.
+ * Only use i0 when using photon over https, based on the assumption that https
+ * maps to http/2 (or later)
+ *
  * @param  {string} pathname The pathname to use
  * @returns {string}          The hostname for the pathname
  */
-function serverFromPathname( pathname ) {
+function serverFromUrlParts( { protocol, pathname } ) {
+	if ( protocol === 'https:' ) {
+		return 'i0.wp.com';
```

Comment:
This will use the hashing algorithm for scheme-relative URLs (`//domain.tld/path`), regardless of whether the current context is secure or not. Is that intentional?

Reply:
hmmmm, no, not intentional. 

Part of me wants to look at the current context, part of me wants to just consider it secure. :)

Reply:
oh, no, it shouldn't. The parts that are passed to serverFromUrlParts are the url being built up, not the original imageUrl.